sshspawner/sshspawner.py

```diff
@@ -200,6 +202,17 @@ def get_remote_user(self, username):
         """Map JupyterHub username to remote username."""
         return username
 
+    def choose_remote_host(self):
```

Comment:
Maybe make this async?  For now we are just doing `random.choice` but if there's a call-out to some service to figure out which host to use we may want to make it async.